packages/p2p-media-loader-core/src/types.d.ts

```diff
@@ -56,8 +56,8 @@ export type Settings = {
   validateP2PSegment?: (url: string, byteRange?: ByteRange) => Promise<boolean>;
 };
 
-export type CoreEventHandlers = {
-  onSegmentLoaded?: (byteLength: number, type: RequestAttempt["type"]) => void;
+export type CoreEventMap = Record<string, (...args: unknown[]) => never> & {
```

Comment:
it didn't work without 
Record<string, (...args: unknown[]) => never> &

Reply:
Fixed